types/wlr_scene.c

```diff
@@ -132,41 +195,142 @@ struct wlr_scene_rect *wlr_scene_rect_create(struct wlr_scene_node *parent,
 	scene_rect->height = height;
 	memcpy(scene_rect->color, color, sizeof(scene_rect->color));
 
+	scene_node_damage_whole(&scene_rect->node);
+
 	return scene_rect;
 }
 
 void wlr_scene_rect_set_size(struct wlr_scene_rect *rect, int width, int height) {
```

Comment:
Now that this adds damage, should it have an early return for no-ops like `wlr_scene_node_set_position` does?

Reply:
Good point, done!